controllers/tempo/tempostack_controller.go

```diff
@@ -241,10 +241,10 @@ func (r *TempoStackReconciler) getStorageConfig(ctx context.Context, tempo v1alp
 		return nil, fmt.Errorf("invalid storage secret: %s", strings.Join(msgs, ", "))
 	}
 
-	return &manifestutils.StorageParams{S3: &manifestutils.S3{
-		Endpoint: string(storageSecret.Data["endpoint"]),
-		Bucket:   string(storageSecret.Data["bucket"]),
-	}}, nil
+	return &manifestutils.StorageParams{
+		S3:           getS3Params(storageSecret),
+		AzureStorage: getAzureParams(storageSecret),
```

Comment:
The storageSecret can be either a S3 secret or an Azure secret, but never both (or am I missing something?).
I'd conditionally either extract the S3 or the Azure params from the secret.

Reply:
I though the same but followed the Loki Operator approach.
The code takes care later of just getting the params associated with the desired storage. I don't have a strong opinion about this. Maybe, I would implement it in the same way you pointed but I maybe it is good to keep the same approach as Loki. I don't  know

Reply:
I took a peek and afaics loki does have a switch based on the type when extracting the storage params from the secret:
https://github.com/grafana/loki/blob/f125682802a556e6b759fce13e1a0c930b57f90f/operator/internal/handlers/internal/storage/secrets.go#L11-L36

Reply:
You're totally right. I'll do the change